pkg/scorecard_test.go

```diff
@@ -341,3 +346,65 @@ func TestExperimentalRunProbes(t *testing.T) {
 		})
 	}
 }
+
+func Test_findConfigFile(t *testing.T) {
+	t.Parallel()
+
+	//nolint:govet
+	tests := []struct {
+		locs      []string
+		desc      string
+		wantFound bool
+	}{
+		{
+			desc:      "scorecard.yml exists",
+			locs:      []string{"scorecard.yml"},
+			wantFound: true,
+		},
+		{
+			desc:      ".scorecard.yml exists",
+			locs:      []string{".scorecard.yml"},
+			wantFound: true,
+		},
+		{
+			desc:      ".github/scorecard.yml exists",
+			locs:      []string{".github/scorecard.yml"},
+			wantFound: true,
+		},
+		{
+			desc:      "multiple configs exist",
+			locs:      []string{"scorecard.yml", ".github/scorecard.yml"},
+			wantFound: true,
+		},
+		{
+			desc:      "no config exists so shouldn't find one",
+			locs:      []string{},
+			wantFound: false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.desc, func(t *testing.T) {
+			t.Parallel()
+			ctrl := gomock.NewController(t)
+			mockRepoClient := mockrepo.NewMockRepoClient(ctrl)
+			mockRepoClient.EXPECT().GetFileReader(gomock.Any()).AnyTimes().DoAndReturn(func(filename string) (io.ReadCloser, error) {
+				if !slices.Contains(tt.locs, filename) {
+					return nil, fmt.Errorf("os.Open: %s", filename)
+				}
+				fullPath := filepath.Join("./testdata", filename)
+				f, err := os.Open(fullPath)
+				if err != nil {
+					return nil, fmt.Errorf("os.Open: %w", err)
+				}
+				return f, nil
```

Comment:
I don't think the slices.Contains block is needed. If the file doesn't exist, os.Open will fail anyway.

So something like  this should be sufficient (wrapping linter is disabled for test funcs)
```go
mockRepoClient.EXPECT().GetFileReader(gomock.Any()).AnyTimes().DoAndReturn(func(filename string) (io.ReadCloser, error) {
	fullPath := filepath.Join("./testdata", filename)
	return os.Open(fullPath)
})
```

Reply:
This is a bit confusing, but my intention was that the test files (which are there in testdata) should only "exist" if the test case def'n says they should, i.e. if the test case def'n includes them in `tt.locs`. The files in testdata are always there though, so os.Open on L397 would always work.